components/admin_console/license_settings/modals/confirm_license_removal_modal.tsx

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useSelector, useDispatch} from 'react-redux';
+
+import {FormattedMessage} from 'react-intl';
+
+import {DispatchFunc} from 'mattermost-redux/types/actions';
+import {GlobalState} from 'mattermost-redux/types/store';
+
+import {isModalOpen} from 'selectors/views/modals';
+
+import GenericModal from 'components/generic_modal';
+import AlertSvg from 'components/common/svg_images_components/alert_svg';
+
+import {ModalIdentifiers} from 'utils/constants';
+
+import {closeModal} from 'actions/views/modals';
+
+import './confirm_license_removal_modal.scss';
+
+type Props = {
+    onExited?: () => void;
+    handleRemove?: (e: any) => Promise<void>;
+}
+
+const ConfirmLicenseRemovalModal: React.FC<Props> = (props: Props): JSX.Element | null => {
+    const dispatch = useDispatch<DispatchFunc>();
+
+    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.CONFIRM_LICENSE_REMOVAL));
+    if (!show) {
+        return null;
+    }
+
+    const handleOnClose = () => {
+        if (props.onExited) {
+            props.onExited();
+        }
+        dispatch(closeModal(ModalIdentifiers.CONFIRM_LICENSE_REMOVAL));
+    };
+
+    const handleRemoval = (e: any) => {
+        if (props.handleRemove) {
+            props.handleRemove(e);
+        }
+        dispatch(closeModal(ModalIdentifiers.CONFIRM_LICENSE_REMOVAL));
+    };
+
+    // Note: DO NOT LOCALISE THESE STRINGS. Legally we can not since the license is in English.
```

Comment:
I don't know if this note applies here. We are localizing in the code. It looks like it might be from copying code from other files. It seems like the only valid instance of this in the in-production codebase is in `components/admin_console/license_settings/ee_license_modal/ee_license_modal.tsx`, and that the other instances in `components/admin_console/license_settings/enterprise_edition/enterprise_edition_left_panel.tsx` and `components/admin_console/license_settings/license_settings.tsx` shouldn't be there, either.

Reply:
good catch! thanks!